pyronear_ds/preprocess/final_merger/final_merger.py

```diff
@@ -0,0 +1,23 @@
+import pandas as pd
+from pandas import DataFrame
+
+from pyronear_ds.preprocess.geographic_data import GeographicData
+
+
+class FinalMerger:
+    def __init__(
+            self, how: str,
+    ):
+        self.how = how
+
+    def get_merged_data(
```

Comment:
Maybe we can join by departement but also by distance. For instance, associate a fire with the closest GHCN station, regardless of if they are in the same department or not : 


`

    def find_closest_station(latitude, longitude):

      stations = df_ghcn_stations_fr.station.values
  
      latitudes = df_ghcn_stations_fr.latitude.values
 
      longitudes = df_ghcn_stations_fr.longitude.values

  
      latitude_0 = df_ghcn_stations_fr.latitude.values[0]
  
      longitude_0 = df_ghcn_stations_fr.longitude.values[0]

 
      min_distance = np.sqrt( (latitude-latitude_0)**2 + (longitude-longitude_0)**2 ) 
  
      reference_station = df_ghcn_stations_fr.station.values[0]
  
      for k in range(len(df_ghcn_stations_fr)):
    
          current_latitude = df_ghcn_stations_fr.latitude.values[k]
    
          current_longitude = df_ghcn_stations_fr.longitude.values[k]
    
          current_distance = np.sqrt( (latitude-current_latitude)**2 + (longitude-current_longitude)**2 )

    
          if current_distance<min_distance:
    
             min_distance = current_distance
    
             reference_station = df_ghcn_stations_fr.station.values[k]

  
          return reference_station`

`

    df_nasa_firms['closest_weather_station'] = df_nasa_firms.apply(lambda row: find_closest_station(row['latitude'],row['longitude']), axis=1)`

then we can join nasa and ghcn on "station + date" combination, we now have a dataset with meteo + fire history + lat/long of the ghcn station

Reply:
WIP